bcipy/simulator/helpers/sim_state.py

```diff
@@ -1,28 +1,129 @@
+
+        return self.state.total_inquiry_count() > self.stop_inq or self.state.target_sentence == self.state.current_sentence or self.state.series_n > 50
+
+    def update(self, evidence: np.ndarray) -> InquiryResult:
+
+        temp_inquiry_result = InquiryResult(target=self.state.target_symbol, time_spent=0, stimuli=self.state.display_alphabet,
+                                            evidence_likelihoods=list(evidence), decision=None)
+
+        # finding out whether max iterations is hit or prob threshold is hit
+        temp_series = copy.deepcopy(self.get_state().series_results)
+        temp_series[-1].append(temp_inquiry_result)
+        is_decidable = any([decider.decide(temp_series[-1]) for decider in self.state.decision_criterion])
+        decision = None
+        # TODO what to do when max inquiry count is reached?
+
+        new_state = self.get_state().__dict__
+        if is_decidable:
+            decision = alphabet()[np.argmax(evidence)]  # deciding the maximum probability symbol TODO abstract
+
+            if decision == self.state.target_symbol:  # correct decision
+                new_state['series_n'] += 1  # TODO abstract out into reset function
+                new_state['series_results'].append([])
+                new_state['inquiry_n'] = 0
+
+                new_state['current_sentence'] += decision
+
+                next_target_symbol_idx = len(new_state['current_sentence'])
+                if next_target_symbol_idx >= len(self.state.target_sentence):
+                    pass
+                else:
+                    new_state['target_symbol'] = self.state.target_sentence[next_target_symbol_idx]
+
+            else:  # wrong decision
+                new_state['series_n'] += 1
+                new_state['series_results'].append([])
+                new_state['inquiry_n'] = 0
+
+        else:
+            new_state['inquiry_n'] += 1
+
+        new_inquiry_result = InquiryResult(target=self.state.target_symbol, time_spent=0, stimuli=self.state.display_alphabet,
+                                           evidence_likelihoods=list(evidence), decision=decision)
+
+        new_state['series_results'][self.state.series_n].append(new_inquiry_result)
+
+        self.state = SimState(**new_state)
+
+        return new_inquiry_result
+
+    def get_state(self) -> SimState:
+        return copy.copy(self.state)
+
+    def mutate_state(self, state_field, state_value) -> SimState:
+        state_dict = self.get_state().__dict__
+        if state_field in state_dict:
+            state_dict[state_field] = state_value
+            self.state = SimState(**state_dict)
+            return self.get_state()
+        else:
+            raise FieldException(f"Cannot find state field {state_field}")
+
+    @staticmethod
+    def initial_state(parameters: Parameters = None) -> SimState:
+        sentence = "HELLO_WORLD"  # TODO abstract out with sim_parameters.json
+        target_symbol = sentence[0]
```

Comment:
We also have a parameter for the number of letters typed so the user can start in the middle of a word or phrase. This can be added later, though.